packages/@aws-cdk/aws-cognito/lib/user-pool.ts

```diff
@@ -822,6 +812,17 @@ export class UserPool extends UserPoolBase {
 
     this.userPoolProviderName = userPool.attrProviderName;
     this.userPoolProviderUrl = userPool.attrProviderUrl;
+
+    if (props.lambdaTriggers) {
+      for (const t of Object.keys(props.lambdaTriggers)) {
+        const trigger = props.lambdaTriggers[t];
+        if (trigger !== undefined) {
+          const name = `${t}${id}`;
+          this.addLambdaPermission(trigger as lambda.IFunction, name);
```

Comment:
I realize this is the solution I commented on the issue but I think there's a better/cleaner solution.

Update the method call here to pass in `scope` - https://github.com/aws/aws-cdk/blob/1aa1588640b34be1dc027f79e3b9dd90e9b8b26f/packages/%40aws-cdk/aws-cognito/lib/user-pool.ts#L842

So it will be -

```ts
fn.addPermission(`${capitalize}Cognito`, {
  principal: new ServicePrincipal('cognito-idp.amazonaws.com'),
  sourceArn: this.userPoolArn,
  scope: this,
});
```